### DISCUSSION TITLE
MAUI & full trust process

### DISCUSSION THREAD

**Discussion by TestTEMV4** [dotnet/maui#13223] [Q&A] [answered]
Hi all,

I am searching for a bit of guidance or at least of information about the run full trust process within MAUI. I have used the FullTrustProcessLauncher in Xamarin with success and as I am currently exploring my options with MAUI to one day fully upgrade my Xamarin apps to MAUI, I stumbled on it while trying to implement it on a MAUI project.

Here is what I tried so far:
1. I tried to use the desktop:Extension on my Winui app package manifest but the value windows.fullTrustProcess in unknown.
2. I tried to use a packaging project. I succeeded in adding my console app exe (the one I want to launch with the full trust process) in the packaging project but can't add my winui app. It is a shame because i had succeeded in adding the desktop:Extension in my packaging project's package manifest.

What am i missing here? Am I maybe just a bit early on this with MAUI?

**Comment by Xyncgas** [comment]
MAUI is Xamarin, with different name

**Reply by TestTEMV4** [reply]
Yes, I know that but the hierarchy of projects is not the same and i can't seem to find a way to implement the run full trust process to launch an exe from the windows app. Did you try it and get it done?

**Comment by TestTEMV4** [comment]
Oooops. My bad.
Just noticed that instead of referencing xmlns:desktop="http://schemas.microsoft.com/appx/manifest/desktop/windows10", I had referenced xmlns:desktop="http://schemas.microsoft.com/appx/manifest/com/windows10"

**Comment by TestTEMV4** [comment]
OK. The com/desktop mistake was definitely part of the problem but it did not solve it entirely.

Here is what I have so far in my maui solution:
1. In the windows package.manifest, I put (I have tried every possible paths I could think of): 
`<Extensions>
        <desktop:Extension Category="windows.fullTrustProcess" Executable="Platforms\Windows\Exe\TestExe.exe"/>
      </Extensions>`
2. My projects in the solution are:
MauApp1
--> Platforms
--> --> Windows
--> --> --> Exe
3. In the csproj of my solution I have
`    <Content Include="Platforms\Windows\Exe\TestExe.exe">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>`
4. In App.xaml.cs I have:
`protected async override void OnLaunched(LaunchActivatedEventArgs args)
    {
        base.OnLaunched(args);

		try
		{
			if (ApiInformation.IsApiContractPresent("Windows.ApplicationModel.FullTrustAppContract", 1, 0))
				await FullTrustProcessLauncher.LaunchFullTrustProcessForCurrentAppAsync();
		}
		catch (Exception ex) 
		{ 
			String s = ex.Message;
		}

    }`

An yet, the exception is raised:
Call context cannot be accessed after call completed. (0x80010117)

Can somebody give me any suggestion?

That process runs with no problem in my xamarin solution. Am I missing something?

**Reply by jameslavery** [reply]
@TestTEMV4 - I'm running into the same problem, did you find a solution?

**Comment by TestTEMV4** [comment, marked as answer]
Hi.

Yes, we did find a solution. Here it is: there is no full trust process per se because you can launch any exe you want in a maui app.

You just need to:
1. Launch the exe
                Process proc = new Process();
                proc.StartInfo.WorkingDirectory = System.IO.Path.Combine (System.IO.Path.GetDirectoryName(aliasPath), "YourExe");
                proc.StartInfo.FileName = "YourExe.exe";
                proc.Start();
2. Reference your exe in your csproj
    "Content Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'" Include="Platforms\Windows\Assets\YourExe\YourExe.exe" />"
 

Have a nice day

**Reply by jameslavery** [reply]
Thanks - that's good news!
